### PR TITLE
chore: Syncs common workflows from the template into main

### PR DESCRIPTION
**Pull request by yorik**

### CODE REVIEW

File: boojum-cuda/Cargo.toml [hunk 1]
```diff
@@ -3,14 +3,15 @@ name = "boojum-cuda"
...
 
 [build-dependencies]
-boojum = { git = "https://github.com/matter-labs/boojum.git", branch = "main" }
+boojum = { git = "https://github.com/matter-labs/era-boojum.git", branch = "main" }
 cmake = "^0"
 itertools = "^0"
 
 [dependencies]
-boojum = { git = "https://github.com/matter-labs/boojum.git", branch = "main" }
+boojum = { git = "https://github.com/matter-labs/era-boojum.git", branch = "main" }
```
Comment (shahar4) [inline boojum-cuda/Cargo.toml:14]: We shouldn't merge with this until we're good to do all of [this](https://matter-labs-workspace.slack.com/archives/C05ATRT8LRE/p1695363194910559). Let's leave this part in it's own PR.

**Review by shahar4 — CHANGES_REQUESTED** [review]

File: boojum-cuda/Cargo.toml (hunk 1 above)
Comment (yorik) [reply boojum-cuda/Cargo.toml:14]: I think era-boojum-cuda isn't used by anyone yet, so it's fine to merge.
Also it can't be used with private repo as dependency.

Comment (shahar4) [reply boojum-cuda/Cargo.toml:14]: @yorik I think in general we should be careful flipping before we make the decision to flip. In any case it's merged now so we can flip.

**Review by shahar4 — APPROVED** [review]
